### DISCUSSION TITLE
Why is next-plugin-google-analytics still experimental?

### DISCUSSION THREAD

**Discussion by jensmeindertsma** [vercel/next.js#12937] [Help] [answered]
I am looking to write a plugin to inject the meta tags needed for a PWA into the Head. I have over 200 lines worth of meta tags and it looks messy in my `_document.tsx`. I have seen the `next-plugin-google-analytics` that injects into the `Head` too. There doesn't seem to be any documentation about how this works ( there is no `index.js` for example ) and the README says it's still experimental. My question is: what's experimental about it? Can I write a plugin just like it and use it? All my plugin needs to do is inject into the Next.js `Head` inside the document.

**Comment by ijjk** [comment, marked as answer]
Hi, the plugins are still experimental while we iterate on the plugins architecture and throughout this iteration a lot of breaking changes will most likely occur and as such should not be relied on in applications yet

**Reply by jensmeindertsma** [reply]
Thanks for the quick response! If I just want to try out creating one, how can I turn it into an installable package? The RFC mentions :

> Next.js will add built-in support for auto-detecting packages with the naming convention @next/plugin-, next-plugin-, or @<scope>/next-plugin-.

Does this already work? So if I create and install a package called `next-plugin-gtag` for example, will this get picked up and used by Next.js automatically? ( assuming I have configured `package.json` too )

**Reply by ijjk** [reply]
The above approach does currently work although is behind [this experimental flag](https://github.com/zeit/next.js/blob/7f28e8bf9bd7dc6a6901968232ad1b28787314c3/packages/next/next-server/server/config.ts#L48) which needs to be enabled in your `next.config.js`. Again, this will most likely be changed dramatically as the feature is iterated on so use is at your own risk

**Reply by jensmeindertsma** [reply]
OK, I'm willing to update it when the feature changes. Thanks for the help and I'll try it out.

**Reply by jensmeindertsma** [reply]
How would I turn this into a NPM package if there is no `index.js` to load from? Is this not necessary?

**Reply by ijjk** [reply]
You can publish it in the same way the `@next/plugin-google-analytics` package is set-up currently. You can name it whatever you like if you pass the package name manually [as shown in this test](https://github.com/zeit/next.js/blob/9c79955047603bd1986c44a888f4f80f3f816d94/test/integration/next-plugins/test/index.test.js#L93-L106) the only requirement is a `nextjs` field in the package's `package.json`

**Reply by jensmeindertsma** [reply]
Thank you so much!
